critiquebrainz/db/users.py

```diff
@@ -70,3 +73,543 @@ def get_many_by_mb_username(usernames):
             else:
                 user["avatar_url"] = gravatar_url(default_gravatar_src)
         return users
+
+
+def get_by_id(user_id):
+    """Get user from user_id
+
+    Args:
+        user_id(uuid): ID of the user
+
+    Returns:
+        Dictionary with the following structure:
+        {
+            "id": (uuid)
+            "display_name": (str)
+            "email": (str)
+            "created": (datetime)
+            "musicbrainz_id": (str)
+            "show_gravatar": (bool)
+            "is_blocked": (bool)
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT id, display_name, email, created,
+                   musicbrainz_id, show_gravatar, is_blocked
+              FROM "user"
+             WHERE id = :user_id
+        """), {
+            "user_id": user_id
+        })
+        row = result.fetchone()
+    return dict(row) if row else None
+
+
+def create(display_name, **kwargs):
+    """Create user using the given details
+
+    Args:
+        display_name(str): display_name of the user,
+        musicbrainz_id(str, optional): musicbrainz_id of user,
+        email(str, optional): email of the user,
+        show_gravatar(bool, optional): whether to show gravatar(default: false)
+        is_blocked(bool, optional): whether user is blocked(default: false)
+
+    Returns:
+        Dictionary with the following structure:
+        {
+            "id": (uuid)
+            "display_name": (str)
+            "email": (str)
+            "created": (datetime)
+            "musicbrainz_id": (str)
+            "show_gravatar": (bool)
+            "is_blocked": (bool)
+        }
+    """
+    id = str(uuid.uuid4())
+    musicbrainz_id = kwargs.pop('musicbrainz_id', None)
+    email = kwargs.pop('email', None)
+    show_gravatar = kwargs.pop('show_gravatar', False)
+    is_blocked = kwargs.pop('is_blocked', False)
+    if(kwargs):
+        raise TypeError('Unexpected **kwargs: %r' % kwargs)
+
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            INSERT INTO "user"
+                 VALUES (:id, :display_name, :email, :created, :musicbrainz_id, :show_gravatar, :is_blocked)
+              RETURNING id
+            """), {
+                "id": id,
+                "display_name": display_name,
+                "email": email,
+                "created": datetime.now(),
+                "musicbrainz_id": musicbrainz_id,
+                "show_gravatar": show_gravatar,
+                "is_blocked": is_blocked
+            })
+        new_id = result.fetchone()[0]
+        user = get_by_id(new_id)
+    return user
+
+
+def get_by_mbid(musicbrainz_id):
```

Comment:
There's one problem with current naming in the schema... It's not exactly correct to call it `musicbrainz_id`, it's actually a username, which is not the same thing as MusicBrainz ID or ID of a user in MusicBrainz.

We can start improving this here by actually calling it `musicbrainz_username` in our database interface. There's no need to rename the row in the schema for now.